lib/GADS/Datum/File.pm

```diff
@@ -68,11 +70,12 @@ after set_value => sub {
             my $old_value   = $self->schema->resultset('Fileval')->find($old[0]); # Only do one fetch here
             my $old_content = $old_value->content;
             my $old_name    = $old_value->name;
-            $changed = 0 if $self->schema->resultset('Fileval')->search({
+            if(my $fl = $self->schema->resultset('Fileval')->search({
                 id      => $values[0],
-                content => $old_content,
                 name    => $old_name
-            })->count;
+            })->next) {
+                $changed = 0 if $fl && $fl->content == $old_content;
```

Comment:
Needs to be ```eq``` otherwise a numerical comparison will be done which might behave not as expected.